source/main.cpp

```diff
@@ -456,7 +463,8 @@ int main(int argc, char *argv[])
         LOG_INFO(TAG, "Jobs is enabled");
         jobs = unique_ptr<JobsFeature>(new JobsFeature());
         jobs->init(resourceManager->getConnection(), listener, config.config);
-        features.push_back(jobs.get());
+        features.add(shared_ptr<Feature>(jobs.get()));
```

Comment:
Bug. unique_ptr is about transfer of ownership. When you call `.get()` on a unique pointer you get a raw pointer to the pointed to object, but the ownership of the object it points to remains with the original unique_ptr. If you were to pass the unique_ptr to FeatureRegistry then the ownership transfers and you can't refer to that object. 

Instead of using `.get()`, what you want to do here is change all of these `unique_ptr` Feature instances to `shared_ptr` and then pass the shared_ptr to `FeatureRegistry::add`. Both main.cpp and FeatureRegistry will share the reference to the same memory. That memory will be deallocated when the reference count for each goes to 0.